x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -102,13 +106,15 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             SSLEngineUtils.extractClientCertificates(logger, threadContext, httpChannel);
         }
 
-        authenticationService.authenticate(maybeWrapRestRequest(request), ActionListener.wrap(authentication -> {
+        final RestRequest wrappedRequest = maybeWrapRestRequest(request);
+        authenticationService.authenticate(wrappedRequest, ActionListener.wrap(authentication -> {
             if (authentication == null) {
                 logger.trace("No authentication available for REST request [{}]", request.uri());
             } else {
                 logger.trace("Authenticated REST request [{}] as {}", request.uri(), authentication);
             }
-            secondaryAuthenticator.authenticateAndAttachToContext(request, ActionListener.wrap(secondaryAuthentication -> {
+            auditTrailService.get().authenticationSuccess(wrappedRequest);
+            secondaryAuthenticator.authenticateAndAttachToContext(wrappedRequest, ActionListener.wrap(secondaryAuthentication -> {
```

Comment:
👍 Using wrappedRequest is in fact a bug fix! Good catch!

Not for this PR, but this prompts me to think: Do we still need audit request body on secondary authc success? It's the current behaviour. I don't think it's necessory. That said, secondary authc probably is not widely used to be a concern.